x/tx/decode/decode_test.go

```diff
@@ -46,9 +53,13 @@ func TestDecode(t *testing.T) {
 	require.NoError(t, err)
 	decoder, err := decode.NewDecoder(decode.Options{
 		SigningContext: signingCtx,
+		ProtoCodec:     mockCodec{},
 	})
 	require.NoError(t, err)
 
+	gogoproto.RegisterType(&bankv1beta1.MsgSend{}, string((&bankv1beta1.MsgSend{}).ProtoReflect().Descriptor().FullName()))
```

Comment:
this is the test fix to make it work.
the alternative was a really hefty refactor

Reply:
brilliant 🙌